### PR TITLE
Cherry pick compliance changes into 0.66-stable

### PR DESCRIPTION
**Pull request by amgleitman**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [x] I am making a change required for Microsoft usage of react-native

## Summary

Bring over compliance changes from the main branch.

Done with the following steps:
* Cherry-pick the following changes: f95f86a1eef, 9e82e970428, 4e19f1f836c, 63075489889, 518e839e011, 87a8f430146, f794feb7bfe
* Run `yarn upgrade` again just in case (some packages in yarn.lock will be ahead of what they are in main simply because new versions were released very recently)

## Changelog

[Internal] [Security] - Security updates

### CODE REVIEW

**Review by harrieshin — APPROVED** [review]

File: android-patches/patches/Hermes/package.json [hunk 1]
```diff
@@ -11,10 +11,3 @@ index 6fc4c801f1..3ad2acfd1c 100644
      "invariant": "^2.2.4",
      "jsc-android": "^250230.2.1",
      "metro-babel-register": "0.66.2",
-@@ -211,4 +211,4 @@
-   "beachball": {
-     "shouldPublish": false
```
Comment (mganandraj) [inline android-patches/patches/Hermes/package.json:16]: @rasaha91 was this change intentional ?

File: repo-config/package.json [hunk 2]
```diff
@@ -21,21 +21,21 @@
...
+    "eslint-config-fb-strict": "^26.0.0",
+    "eslint-config-fbjs": "^3.1.1",
+    "eslint-config-prettier": "^8.3.0",
+    "eslint-plugin-babel": "^5.3.1",
+    "eslint-plugin-eslint-comments": "^3.2.0",
+    "eslint-plugin-flowtype": "^7.0.0",
+    "eslint-plugin-jest": "^25.2.4",
+    "eslint-plugin-jsx-a11y": "^6.4.1",
+    "eslint-plugin-prettier": "^4.0.0",
+    "eslint-plugin-react": "^7.26.1",
```
Comment (mganandraj) [inline repo-config/package.json:35]: Looks like a downgrade ?

**Review by mganandraj — APPROVED** [review]
